### PR TITLE
getBySlug

### PR DESCRIPTION
**Pull request by timbur**
also: highwinds purging

### CODE REVIEW

File: src/migrations.js [hunk 1]
```diff
@@ -5,7 +5,14 @@ export default {
       table: 'pages',
       indexes: [
         'applicationId',
-        'organizationId'
+        'organizationId',
+        'slug'
+      ],
+      compoundIndexes: [
+        {
+          name: 'link_pageSlug_applicationId',
```
Comment (tom-james-watson) [inline src/migrations.js]: I think the convention would be `link_page_slug_application`

File: test/integration/routes.js [hunk 2]
```diff
@@ -78,14 +86,49 @@ describe('Routes', function() {
...
+
+    var res = await server
+      .get(`/api/${process.env.API_VERSION}/pages/slug/${pageSlug}`)
+      .set('organizationid', orgId)
+      .set('applicationid', appId)
+
+    expect(res.body.status).to.equal(200)
+    expect(res.body.pkg.id).to.equal(pageId)
+    expect(res.body.pkg.title).to.equal(pkg.title)
+    expect(res.body.pkg.slug).to.equal(pkg.slug)
```
Comment (tom-james-watson) [inline test/integration/routes.js]: You can check that your hook works properly by spying on the queue, like is done here https://github.com/f12/paradigm-documents/blob/master/test/integration/routes/document.js#L43

Comment (timbur) [reply test/integration/routes.js]: Nice.  I'll add something there.

File: src/migrations.js (hunk 1 above)
Comment (timbur) [reply src/migrations.js]: I modeled this after https://github.com/f12/paradigm-channels/blob/b15c7ff233489e50e23f74053b63ee6f7ce4914b/src/migrations.js#L17

Comment (tom-james-watson) [reply src/migrations.js]: 👍

**Pull request merged**